runtime/lua/vim/lsp/log.lua

```diff
@@ -32,6 +32,17 @@ do
 
   vim.fn.mkdir(vim.fn.stdpath('cache'), "p")
   local logfile = assert(io.open(logfilename, "a+"))
+
+	local log_info = vim.loop.fs_stat(logfilename)
+	if log_info and log_info.size > 1e9 then
+		local warn_msg = string.format(
+			"Language server client log at %s has reached %d KB in size",
+			logfilename,
+			log_info.size / 1000
```

Comment:
if the variable part of a message must go in the middle of a message, try to put the longest part (filename in this case) at the end. 

```suggestion
			"LSP client log is big (%d MB): %s",
			log_info.size / (1000 * 1000),
			logfilename,
```